### PR TITLE
EBS encryption

### PR DESCRIPTION
**Pull request by daxmc99**
Addresses https://github.com/rancher/rancher/issues/21552

### CODE REVIEW

**Comment by bensonarokiadoss** [comment]
@daxmc99 will the console UI and API changes will be addressed separated or picked automatically as part of this change

**Comment by rmweir** [comment]
as per @bensonarokiadoss, they are done separately. With that being said, @daxmc99 can you open a UI issue for this?

**Comment by dramich** [comment]
This can't be merged until after 2.3 branch is cut.

File: drivers/eks/eks_driver.go [hunk 1]
```diff
@@ -317,6 +325,7 @@ func getStateFromOptions(driverOptions *types.DriverOptions) (state, error) {
 	state.AMI = options.GetValueFromDriverOptions(driverOptions, types.StringType, "ami").(string)
 	state.AssociateWorkerNodePublicIP, _ = options.GetValueFromDriverOptions(driverOptions, types.BoolPointerType, "associate-worker-node-public-ip", "associateWorkerNodePublicIp").(*bool)
 	state.KeyPairName = options.GetValueFromDriverOptions(driverOptions, types.StringType, "keyPairName").(string)
+	state.EbsEncryption, _ = options.GetValueFromDriverOptions(driverOptions, types.BoolPointerType, "ebs-encryption", "EbsEncryption").(bool)
```
Comment (dramich) [inline drivers/eks/eks_driver.go]: Drop the `_`, let's try and keep the formatting the same with all the other options.

File: drivers/eks/templates.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package eks
 
+// This are the cloudformation templates used for eks clusters, when making edits here ensure that view whitespace is enabled
```
Comment (dramich) [inline drivers/eks/templates.go]: `// These are the CloudFormation templates used for EKS clusters, when making edits here ensure the whitespace is correct.``

File: drivers/eks/eks_driver.go [hunk 3]
```diff
@@ -264,6 +265,13 @@ func (d *Driver) GetDriverCreateOptions(ctx context.Context) (*types.DriverFlags
 		Usage:   "The kubernetes master version",
 		Default: &types.Default{DefaultString: "1.13"},
 	}
+	driverFlag.Options["ebs-encryption"] = &types.Flag{
+		Type:  types.BoolType,
+		Usage: "Enables ebs encryption of worker nodes",
```
Comment (dramich) [inline drivers/eks/eks_driver.go]: ebs > EBS

**Comment by daxmc99** [comment]
UI issue has been filed https://github.com/rancher/rancher/issues/22633

**Pull request merged**